### PR TITLE
Allows Skills to Store Info About Blogs, Tutorials, Webpages...

### LINKED ISSUE #5: Skills need to have a the ability to save/read links to webpages, blogs, and tutorials
When performing a get on a `Skill`, this is what I'm hoping to receive.
For example:
```
Skill{
  Name: "Java"
  Id: id_string
  Type: "compiled"
  Webpage: Link {Name: "Java Website", Url: "https://www.java.com/en/"}
  Blogs: []Link{Link{Name:"Java", Url: "http://www.javaworld.com/blog/java-101/"}}
  Tutorials: []Link{}
}
```

The Link type would be something like this.  I'm anticipating there will need to be a separate query to inject the links into a skill object
```
Link{
  Name string
  Url string
  Skill_Id string
  LinkType string //blog, tutorial, webpage
}
```

### PR DESCRIPTION
**Pull request by Hopding**
Fixes #5

### CODE REVIEW

File: controller/skills.go [hunk 1]
```diff
@@ -146,14 +146,20 @@ func (c *SkillsController) addSkill() error {
...
-			ErrorMsg: "Invalid JSON body in request:\n\t" + fmt.Sprint(body),
+			ErrorMsg: err.Error() + "\nInvalid request with following JSON body:\n\t" + string(body[:]),
 		}
 	}
 	if !model.IsValidSkillType(skill.SkillType) {
 		return &errors.InvalidSkillTypeError{
 			ErrorMsg: "Invalid Skill Type: %s" + skill.SkillType,
 		}
 	}
+	skill.Blogs = []model.Link{
```
Comment (thebho) [inline controller/skills.go]: Why is this hard coded into the skills controller?

File: model/skill.go [hunk 2]
```diff
@@ -13,11 +16,16 @@ const (
 	DatabaseSkillType      = "database"
 )
 
-func NewSkill(id, name, skillType string) Skill {
+func NewSkill(id, name, skillType string,
```
Comment (thebho) [inline model/skill.go]: I think we may want to keep this the old way, and add a different constructor for `NewSkillWithLinks` and `(s Skill)AddLink` with ( link `Link`,  linkType `string` ) arguments.

**Review by thebho — CHANGES_REQUESTED** [review]

File: controller/skills.go (hunk 1 above)
Comment (Hopding) [reply controller/skills.go]: That is a testing remnant that I forgot to remove. It should not be there.

File: model/skill.go (hunk 2 above)
Comment (Hopding) [reply model/skill.go]: That seems reasonable 👍

**Comment by Hopding** [comment]
Changes made. Should be good to go 👍

**Review by thebho — APPROVED** [review]

**Pull request merged**